src/main/java/redis/clients/jedis/BinaryJedis.java

```diff
@@ -110,9 +110,6 @@ private void initializeFromClientConfig(JedisClientConfig config) {
       }
     } catch (JedisException je) {
       try {
-        if (isConnected()) {
-          quit();
-        }
         disconnect();
```

Comment:
```suggestion
        setBroken();
        disconnect();
```
Jedis4 actively called `setBroken()` in the `quit()` command, we can also add it here.

Reply:
@yangbodong22011 We don't have `setBroken()` in Jedis 3.x.

Reply:
ok, no problem then.